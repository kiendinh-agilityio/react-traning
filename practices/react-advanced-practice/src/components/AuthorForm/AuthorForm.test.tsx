import { render, fireEvent, screen, act } from '@testing-library/react';
import AuthorForm from '.';
import { PROFILE_AUTHORS } from '@/mocks';
import { Author } from '@/types';

const mockProps = {
  isUpdate: false,
  selectedAuthor: PROFILE_AUTHORS,
  closeModal: jest.fn(),
  onChange: jest.fn(),
  onSubmit: jest.fn(),
};

describe('AuthorForm', () => {
  it('should render correctly with default props', () => {
    const { container } = render(<AuthorForm {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly in update mode', () => {
    const updateProps = { ...mockProps, isUpdate: true };
    render(<AuthorForm {...updateProps} />);
    expect(screen.getByText('EDIT AUTHOR')).toBeInTheDocument();
  });

  it('should render correctly with empty selectedAuthor', () => {
    const emptyAuthorProps = { ...mockProps, selectedAuthor: {} as Author };
    render(<AuthorForm {...emptyAuthorProps} />);
    expect(screen.getByPlaceholderText('Please enter name')).toBeInTheDocument();
  });

  it('should handle field changes correctly', async () => {
    render(<AuthorForm {...mockProps} />);
    const nameInput = screen.getByPlaceholderText('Please enter name');

    await act(async () => {
      fireEvent.change(nameInput, { target: { value: 'New Author Name' } });
    });

    expect(mockProps.onChange).toHaveBeenCalled();
  });

  it('should validate date correctly', async () => {
    render(<AuthorForm {...mockProps} />);
    const dateInput = screen.getByPlaceholderText('');

    await act(async () => {
      fireEvent.change(dateInput, { target: { value: 'invalid-date' } });
    });

    expect(mockProps.onChange).toHaveBeenCalled();
  });

  it('should disable submit button correctly when not dirty in update mode', () => {
    const updateProps = { ...mockProps, isUpdate: true };
    render(<AuthorForm {...updateProps} />);
    const submitButton = screen.getByText('Save');
    expect(submitButton).toBeDisabled();
  });

  it('should call closeModal when cancel button is clicked', async () => {
    render(<AuthorForm {...mockProps} />);
    const cancelButton = screen.getByText('Cancel');

    await act(async () => {
      fireEvent.click(cancelButton);
    });

    expect(mockProps.closeModal).toHaveBeenCalled();
  });
});
