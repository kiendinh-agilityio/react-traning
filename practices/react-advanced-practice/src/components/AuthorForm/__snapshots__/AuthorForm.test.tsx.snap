// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorForm should render correctly in update mode 1`] = `
<div>
  <h2
    class="rt-Heading rt-r-size-6 rt-r-weight-bold text-lg font-bold text-dark font-bold text-center"
  >
    EDIT AUTHOR
  </h2>
  <form
    class="flex flex-col gap-6"
  >
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Name
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="name"
            placeholder="Please enter name"
            type="text"
            value="Ken"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Email
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="email"
            placeholder="Please enter email address"
            type="email"
            value="example@gmail.com"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Avatar
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="avatarUrl"
            placeholder="Please enter link image"
            type="text"
            value="https://phimtat.vn/wp-content/uploads/2023/11/avt-3d.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Employed
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="date"
            placeholder=""
            type="date"
            value="26/10/24"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Grid rt-r-gtc-3 gap-5"
    >
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Roles
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="roles"
            name="roles"
          >
            <option
              value="Manager"
            >
              Manager
            </option>
            <option
              value="Programmer"
            >
              Programmer
            </option>
            <option
              value="Executive"
            >
              Executive
            </option>
            <option
              value="Designer"
            >
              Designer
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Positions
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="position"
            name="position"
          >
            <option
              value="Organization"
            >
              Organization
            </option>
            <option
              value="Developer"
            >
              Developer
            </option>
            <option
              value="Projects"
            >
              Projects
            </option>
            <option
              value="UI/UX Design"
            >
              UI/UX Design
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Status
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="status"
            name="status"
          >
            <option
              value="Active"
            >
              Active
            </option>
            <option
              value="Inactive"
            >
              Inactive
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-ai-center rt-r-jc-center gap-6 mt-6"
    >
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer primary opacity-50 cursor-not-allowed
   w-[150px]"
        data-accent-color=""
        data-disabled="true"
        disabled=""
      >
        Save
      </button>
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer tertiary
   w-[150px]"
        data-accent-color=""
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;

exports[`AuthorForm should render correctly with default props 1`] = `
<div>
  <h2
    class="rt-Heading rt-r-size-6 rt-r-weight-bold text-lg font-bold text-dark font-bold text-center"
  >
    ADD AUTHOR
  </h2>
  <form
    class="flex flex-col gap-6"
  >
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Name
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="name"
            placeholder="Please enter name"
            type="text"
            value="Ken"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Email
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="email"
            placeholder="Please enter email address"
            type="email"
            value="example@gmail.com"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Avatar
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="avatarUrl"
            placeholder="Please enter link image"
            type="text"
            value="https://phimtat.vn/wp-content/uploads/2023/11/avt-3d.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Employed
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="date"
            placeholder=""
            type="date"
            value="26/10/24"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Grid rt-r-gtc-3 gap-5"
    >
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Roles
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="roles"
            name="roles"
          >
            <option
              value="Manager"
            >
              Manager
            </option>
            <option
              value="Programmer"
            >
              Programmer
            </option>
            <option
              value="Executive"
            >
              Executive
            </option>
            <option
              value="Designer"
            >
              Designer
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Positions
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="position"
            name="position"
          >
            <option
              value="Organization"
            >
              Organization
            </option>
            <option
              value="Developer"
            >
              Developer
            </option>
            <option
              value="Projects"
            >
              Projects
            </option>
            <option
              value="UI/UX Design"
            >
              UI/UX Design
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Status
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="status"
            name="status"
          >
            <option
              value="Active"
            >
              Active
            </option>
            <option
              value="Inactive"
            >
              Inactive
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-ai-center rt-r-jc-center gap-6 mt-6"
    >
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer primary
   w-[150px]"
        data-accent-color=""
      >
        Submit
      </button>
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer tertiary
   w-[150px]"
        data-accent-color=""
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;

exports[`AuthorForm should render correctly with empty selectedAuthor 1`] = `
<div>
  <h2
    class="rt-Heading rt-r-size-6 rt-r-weight-bold text-lg font-bold text-dark font-bold text-center"
  >
    ADD AUTHOR
  </h2>
  <form
    class="flex flex-col gap-6"
  >
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Name
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="name"
            placeholder="Please enter name"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Email
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="email"
            placeholder="Please enter email address"
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Avatar
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="avatarUrl"
            placeholder="Please enter link image"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Employed
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="date"
            placeholder=""
            type="date"
            value="2025-03-03"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Grid rt-r-gtc-3 gap-5"
    >
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Roles
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="roles"
            name="roles"
          >
            <option
              value="Manager"
            >
              Manager
            </option>
            <option
              value="Programmer"
            >
              Programmer
            </option>
            <option
              value="Executive"
            >
              Executive
            </option>
            <option
              value="Designer"
            >
              Designer
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Positions
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="position"
            name="position"
          >
            <option
              value="Organization"
            >
              Organization
            </option>
            <option
              value="Developer"
            >
              Developer
            </option>
            <option
              value="Projects"
            >
              Projects
            </option>
            <option
              value="UI/UX Design"
            >
              UI/UX Design
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Status
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="status"
            name="status"
          >
            <option
              value="Active"
            >
              Active
            </option>
            <option
              value="Inactive"
            >
              Inactive
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-ai-center rt-r-jc-center gap-6 mt-6"
    >
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer primary
   w-[150px]"
        data-accent-color=""
      >
        Submit
      </button>
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer tertiary
   w-[150px]"
        data-accent-color=""
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;

exports[`AuthorForm should render correctly with errors 1`] = `
<div>
  <h2
    class="rt-Heading rt-r-size-6 rt-r-weight-bold text-lg font-bold text-dark font-bold text-center"
  >
    ADD AUTHOR
  </h2>
  <form
    class="flex flex-col gap-6"
  >
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Name
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="name"
            placeholder="Please enter name"
            type="text"
            value="invalid name"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Email
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="email"
            placeholder="Please enter email address"
            type="email"
            value="invalid email"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Avatar
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="avatarUrl"
            placeholder="Please enter link image"
            type="text"
            value="invalid avatarUrl"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column gap-1.5"
    >
      <label
        class="rt-Text rt-r-weight-regular text-base text-dark"
      >
        Employed
      </label>
      <div
        class="rt-Box w-full"
      >
        <div
          class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
        >
          <input
            class="w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
            name="date"
            placeholder=""
            type="date"
            value="invalid date"
          />
        </div>
      </div>
    </div>
    <div
      class="rt-Grid rt-r-gtc-3 gap-5"
    >
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Roles
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="roles"
            name="roles"
          >
            <option
              value="Manager"
            >
              Manager
            </option>
            <option
              value="Programmer"
            >
              Programmer
            </option>
            <option
              value="Executive"
            >
              Executive
            </option>
            <option
              value="Designer"
            >
              Designer
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Positions
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="position"
            name="position"
          >
            <option
              value="Organization"
            >
              Organization
            </option>
            <option
              value="Developer"
            >
              Developer
            </option>
            <option
              value="Projects"
            >
              Projects
            </option>
            <option
              value="UI/UX Design"
            >
              UI/UX Design
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="rt-Flex rt-r-fd-column gap-1.5 text-dark font-regular"
      >
        <label>
          Status
        </label>
        <div
          class="rt-Flex relative"
        >
          <select
            class="appearance-none px-5 py-[15px] border w-full rounded-[15px] focus:outline-none mb-1.5 border-input"
            id="status"
            name="status"
          >
            <option
              value="Active"
            >
              Active
            </option>
            <option
              value="Inactive"
            >
              Inactive
            </option>
          </select>
          <span
            class="rt-Box absolute bottom-1/2 right-[20px]"
          >
            <svg
              fill="none"
              height="7"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.243 6.329 0 2.086 1.415.672 4.243 3.5 7.071.67l1.415 1.415-4.243 4.243Z"
                fill="#C4C4C4"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-ai-center rt-r-jc-center gap-6 mt-6"
    >
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer primary
   w-[150px]"
        data-accent-color=""
      >
        Submit
      </button>
      <button
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl cursor-pointer tertiary
   w-[150px]"
        data-accent-color=""
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
